include/event_manager.h

```diff
@@ -147,52 +144,142 @@ extern "C" {
  */
 #define EVENT_SUBMIT(event) _event_submit(&event->header)
 
+/**
+ * @brief Register event hook after the event manager is initialized
+ *
+ * The event hook called after the event manager is initialized to provide some additional
+ * initialization of the modules that depends on it.
+ * The hook function should have a form `int hook(void)`.
+ * If the initialization hook returns non-zero value the initialization process is interrupted.
+ *
+ * @param hook_fn Hook function
+ */
+#define EVENT_MANAGER_HOOK_POSTINIT_REGISTER(hook_fn) \
+	_EVENT_MANAGER_HOOK_POSTINIT_REGISTER(hook_fn, _EM_SUBS_PRIO_ID(_EM_SUBS_PRIO_NORMAL))
 
-/** @brief Initialize the Event Manager.
+/**
+ * @brief Register hook called on event submission. The hook would be called first.
  *
- * @retval 0 If the operation was successful. Error values can be added by overwriting
- *         event_manager_trace_event_init function.
+ * The event hook called when the event is submitted.
+ * The hook function should have a form `void hook(const struct event_header *eh)`.
+ * The macro makes sure that the hook provided here is called first.
+ * Only one hook can be registered with this macro.
+ *
+ * @param hook_fn Hook function
  */
-int event_manager_init(void);
+#define EVENT_HOOK_ON_SUBMIT_REGISTER_FIRST(hook_fn)                                        \
+	const struct {} __event_hook_on_submit_first_sub_redefined = {};                    \
+	_EVENT_HOOK_ON_SUBMIT_REGISTER(hook_fn, _EM_SUBS_PRIO_ID(_EM_MARKER_FIRST_ELEMENT))
```

Comment:
I think we should use `_EM_SUBS_PRIO_ID` only with `_EM_SUBS_PRIO_EARLY` and `_EM_SUBS_PRIO_NORMAL`

Reply:
It should not be used with `_EM_MARKER_...`. `_EM_MARKER_...` is used as it is without wrapper

Reply:
Also in other places

Reply:
For the first comment - I do not see a sense for that. We have predefined first and last markers. It would be really misleading when in first and last macros we would use early and late priorites. Why only Normal and Early? For the sake of Event Manager Tracer we would need to register first for the event preprocessing and last for the event postprocessing.

Reply:
Fixed.

Reply:
The `_EM_MARKER_FIRST_ELEMENT` is still wrapped with `_EM_SUBS_PRIO_ID()`. As I mentioned in previous comment: `_EM_MARKER_...` is used as it is without wrapping it with `_EM_SUBS_PRIO_ID`.
`_EM_SUBS_PRIO_ID` doc clearly states that `The level must be between 00 and 99.`

Reply:
Pushed now - sorry.